### ISSUE TITLE
Python 3.11 - getdefaultlocale() deprecated

### ISSUE THREAD

**Issue by vsalvino** [pytest-dev/pytest-nunit#67]
Running this package on Python 3.11 yields deprecation warnings:

```
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/site-packages/pytest_nunit/nunit.py:119: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
    language_code = locale.getdefaultlocale()[0]
```

See: 
* https://github.com/python/cpython/issues/90817
* https://docs.python.org/release/3.11.0/library/locale.html#locale.getdefaultlocale

(I will try to provide a fix at some point if nobody else takes it up... logging the issue here for memory's sake)

**Comment by jamesmyatt** [comment]
I'm not clear about what the fix for this is. Is it just to call `locale.getlocale()` instead? On my machine, that returns the locale in a different form.

**Comment by vsalvino** [comment]
After looking into it, I don't actually know what the correct upgrade path is. Admittedly I don't know much about how locales work. Stumbled across this bug which might be worth watching: https://github.com/python/cpython/issues/82986

**Comment by MalcolmTyler** [comment]
Changing :
 **language_code = locale.getdefaultlocale()[0]**
 to
 **language_code = locale.getlocale()[0]**
 worked for me, they both return the same only the first is being deprecated so gives a warning

**Comment by mjpieters** [comment]
> Changing : **language_code = locale.getdefaultlocale()[0]** to **language_code = locale.getlocale()[0]** worked for me, they both return the same only the first is being deprecated so gives a warning

They may give the same for you, but that's not the case for all platforms. See python/cpython#82986 as to why this could be problematic on Windows.

**Comment by MalcolmTyler** [comment]
> > Changing : **language_code = locale.getdefaultlocale()[0]** to **language_code = locale.getlocale()[0]** worked for me, they both return the same only the first is being deprecated so gives a warning
> 
> They may give the same for you, but that's not the case for all platforms. See [python/cpython#82986](https://github.com/python/cpython/issues/82986) as to why this could be problematic on Windows.
> 
> Please read the whole discussion here, as this was _already covered_.

Fair point, we were seeing the deprecation warning when using pytest and it was only MAC users that were seeing it.

**Comment by mathrick** [comment]
`locale.setlocale(locale.LC_CTYPE, None)` will return a string describing the current locale. Be aware however that it also includes the encoding, ie. `en_GB.UTF-8`, rather than just `en_GB`.

**Comment by alecglen** [comment]
Does anyone have a temp solution to hide these from the xml output? I'd love to not have 900 lines of this per run in my log.

**Comment by mathrick** [comment]
> Does anyone have a temp solution to hide these from the xml output? I'd love to not have 900 lines of this per run in my log.

Add the following in your `pytest.ini`:
```ini
filterwarnings =
    ignore:Use setlocale:DeprecationWarning:pytest_nunit
```

**Comment by tonybaloney** [comment]
> > Changing : **language_code = locale.getdefaultlocale()[0]** to **language_code = locale.getlocale()[0]** worked for me, they both return the same only the first is being deprecated so gives a warning
> 
> They may give the same for you, but that's not the case for all platforms. See [python/cpython#82986](https://github.com/python/cpython/issues/82986) as to why this could be problematic on Windows.
> 
> Please read the whole discussion here, as this was _already covered_.

I'm going to mute the deprecation warning because `getlocale` seems fraught with complexity and unresolved issues after reading this and many other related discussions

**Comment by tonybaloney** [comment]
Fixed in 1.0.5
